src/layers/conv.jl

```diff
@@ -118,26 +121,40 @@ function (l::GCNConv)(g::GNNGraph,
         end
     end
     Dout, Din = size(l.weight)
-    if Dout < Din
-        # multiply before convolution if it is more convenient, otherwise multiply after
+    if Dout < Din && !(g isa GNNHeteroGraph)
         x = l.weight * x
     end
-    if edge_weight !== nothing
-        d = degree(g, T; dir = :in, edge_weight)
+    if g isa GNNHeteroGraph
+        d = degree(g, edge_t, T; dir = :in)
     else
-        d = degree(g, T; dir = :in, edge_weight = l.use_edge_weight)
+        if edge_weight !== nothing
+            d = degree(g, T; dir = :in, edge_weight)
+        else
+            d = degree(g, T; dir = :in, edge_weight = l.use_edge_weight)
+        end
     end
     c = norm_fn(d)
-    x = x .* c'
-    if edge_weight !== nothing
-        x = propagate(e_mul_xj, g, +, xj = x, e = edge_weight)
-    elseif l.use_edge_weight
-        x = propagate(w_mul_xj, g, +, xj = x)
+    if xi != xj
```

Comment:
```suggestion
    if xi !== xj
```
we should check identity, not equality